maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java

```diff
@@ -158,7 +158,7 @@ public void testCalculateDegreeOfConcurrency() {
         int cpus = Runtime.getRuntime().availableProcessors();
         assertEquals((int) (cpus * 2.2), cli.calculateDegreeOfConcurrency("2.2C"));
         assertEquals(1, cli.calculateDegreeOfConcurrency("0.0001C"));
-        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("2.C"));
+        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("2.xC"));
```

Comment:
I don't understand why this changed here

Reply:
When changing the related code, I found out that `2.C` was rejected early, although fully supported, so the test ended up failing after removing the early test which was using commons-lang `Validate`, as this part is completely duplicated by the exception thrown by the `Float.parseFloat` which also validates the syntax.

Reply:
OK, so it's an extra test. That's fine, but I suggest also keeping 

```
assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("2.C"));
```

Reply:
I can't, because that one now passes, as `2.` is a legit float number according to the `Float.parseFloat` which is now used to validate the syntax.

Reply:
It's a behaviour change for sure, whether this is considered a bug can be discussed I suppose.  I'll raise a JIRA for that one and rebase on top of it.

Reply:
I've splitted in two commits (not to be squashed) and two JIRA issues...